examples/contact/constraint_twist.cpp

```diff
@@ -13,20 +13,95 @@
 
 #include "mfem.hpp"
 
+// ContinuationSolver headers
+#include "problems/Problems.hpp"
+#include "solvers/HomotopySolver.hpp"
+#include "utilities.hpp"
+
 #include "serac/serac.hpp"
 
+
+static constexpr int dim = 3;
+static constexpr int disp_order = 1;
+
+using VectorSpace = serac::H1<disp_order, dim>;
+
+using DensitySpace = serac::L2<disp_order - 1>;
+
+using SolidMaterial = serac::solid_mechanics::NeoHookeanWithFieldDensity;
+using SolidWeakFormT = serac::SolidWeakForm<disp_order, dim, serac::Parameters<DensitySpace>>;
+
+enum FIELD
+{
+  DISP = SolidWeakFormT::DISPLACEMENT,
+  VELO = SolidWeakFormT::VELOCITY,
+  ACCEL = SolidWeakFormT::ACCELERATION,
+  DENSITY = SolidWeakFormT::NUM_STATES
+};
+
+/* NLMCP of the form
+ * 0 <= x \perp F(x, y) >= 0
+ *              Q(x, y)  = 0
+ * Here, F and x are both 0-dimensional
+ * and   Q(x, y) = [ r(u) + (dc/du)^T l]
+ *                 [-c(u)]
+ *            y  = [ u ]
+ *                 [ l ]
+ * we use the approximate Jacobian
+ *       dQ/dy \approx [ dr/du     (dc/du)^T]
+ *                     [-dc/du        0 ]
+ * we note that the sign-convention with regard to "c" is important
+ * as  the approximate Jacobian is positive semi-definite when dr/du is
+ * and thus the NLMC problem is guaranteed to be semi-monotone.
+ */
+template <typename SolidWeakFormType>
+class TiedContactProblem : public GeneralNLMCProblem {
+ protected:
+  mfem::HypreParMatrix* dFdx;
+  mfem::HypreParMatrix* dFdy;
+  mfem::HypreParMatrix* dQdx;
+  mfem::HypreParMatrix* dQdy;
+  HYPRE_BigInt* uOffsets;
+  HYPRE_BigInt* cOffsets;
+  int dimu;
+  int dimc;
+  int dimcglb;
+  mfem::Array<int> y_partition;
+  std::vector<serac::FieldPtr> contact_states;
+  std::vector<serac::FieldPtr> all_states;
+  std::shared_ptr<SolidWeakFormType> weak_form;
+  std::unique_ptr<serac::FiniteElementState> shape_disp;
+  std::shared_ptr<serac::Mesh> mesh;
+  std::shared_ptr<serac::ContactConstraint> constraints;
+  double time = 0.0;
+  double dt = 0.0;
+  std::vector<double> jacobian_weights = {1.0, 0.0, 0.0, 0.0};
+
+ public:
+  TiedContactProblem(std::vector<serac::FieldPtr> contact_states_, std::vector<serac::FieldPtr> all_states_,
```

Comment:
Our project naming conventions are class members variables end in `_`. Please swap the parameters to not have the trailing underscore but have the member variables above to have them.